### PR TITLE
Add types for '@storybook/vue' package

### PR DESCRIPTION
**Pull request by pntgupta**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by pntgupta** [comment]
There is a common mistake in PR, I have used "storiesOf<T>(kind: string, module: NodeModule): Story & T". Please suggest if there is any other better way of doing the same.

**Comment by pntgupta** [comment]
@johnnyreilly Can you please help. I am importing 'vue' but since vue isn't available in DefinitelyTyped so travis CI build is failing.

**Comment by andy-ms** [comment]
To depend on `vue` you need a `package.json` file like `vue-scrollto` has.

File: types/storybook__vue/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    name: string;
+    render: StoryFunction;
+}
+
+export function addDecorator(decorator: StoryDecorator): void;
+export function configure(loaders: () => void, module: NodeModule): void;
+export function getStorybook(): StoryStore[];
+export function setAddon(addon: Addon): void;
+export function storiesOf(kind: string, module: NodeModule): Story;
+export function storiesOf<T>(kind: string, module: NodeModule): Story & T;
```
Comment (andy-ms) [inline types/storybook__vue/index.d.ts]: This overload isn't necessary -- one can just cast the return value to any subtype of `Story`.

File: types/storybook__vue/tslint.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-unnecessary-generics": false
```
Comment (andy-ms) [inline types/storybook__vue/tslint.json]: Remove this -- see above comment.

**Comment by pntgupta** [comment]
@andy-ms Thanks for the help. I have updated the PR.

**Pull request merged**